### PR TITLE
Update Nominatim.php

### LINKED ISSUE #853: Provide valid User-Agent for Nominatim
Per https://github.com/eileenmcnaughton/org.wikimedia.geocoder/issues/1 it has been reported to us that 

"The nominatim.openstreetmap.org states that users must "provide a valid HTTP Referer or User-Agent identifying the application (stock User-Agents as set by http libraries will not do)."

The User-Agent is currently set to the default GuzzleHttp one. This is in violation of the usage policy."

Is it currently possible to control the user-agent with the Nominatim plugin to avoid this?

### PR DESCRIPTION
**Pull request by jbelien**
Add `User-Agent` and `Referer` headers.
Fix #853

### CODE REVIEW

**Comment by jbelien** [comment]
Travis failed for PHP7 deps=low but it seems to be caused by GeoIP provider and not be related to Nominatim provider.

File: src/Provider/Nominatim/Nominatim.php [hunk 1]
```diff
@@ -210,7 +210,16 @@ private function executeQuery(string $url, string $locale = null): string
...
         }
 
-        return $this->getUrlContents($url);
+        $request = $this->getRequest($url);
+
+        if (isset($_SERVER['HTTP_USER_AGENT'])) {
+            $request = $request->withHeader('User-Agent', $_SERVER['HTTP_USER_AGENT']);
+        }
+        if (isset($_SERVER['HTTP_REFERER'])) {
+            $request = $request->withHeader('Referer', $_SERVER['HTTP_REFERER']);
```
Comment (Nyholm) [inline src/Provider/Nominatim/Nominatim.php]: Is Referer need?

**Comment by pnorman** [comment]
This PR *allows* the referer to be set, but doesn't appear to actually set it, so does it fix #853 by itself?

**Comment by jbelien** [comment]
@Nyholm Yeah, I also asked myself the same question ...
And you're probable right, not a good thing to use global variables but the User-Agent or Referer **needs** to be set : 
> Provide a valid HTTP Referer or User-Agent identifying the application (stock User-Agents as set by http libraries will not do).

@pnorman @Nyholm What would you the best way to do it you think ?

File: src/Provider/Nominatim/Nominatim.php (hunk 1 above)
Comment (jbelien) [reply src/Provider/Nominatim/Nominatim.php]: User-Agent **or** Referer needs to be set :
> Provide a valid HTTP Referer or User-Agent identifying the application (stock User-Agents as set by http libraries will not do).

**Comment by Nyholm** [comment]
I think the best is to not use Referer and to let the user configure User-Agent with constructor injection.

**Comment by jbelien** [comment]
Constructor (and tests) updated !
I also added the [Nominatim Usage Policy link](https://operations.osmfoundation.org/policies/nominatim/) to the README file.

---
> This PR allows the referer to be set, but doesn't appear to actually set it, so does it fix #853 by itself?

@pnorman I checked, the referer is actually set !

**Comment by Nyholm** [comment]
Excellent. Thank you

**Pull request merged**